tests/cnf/ran/gitopsztp/tests/ztp-siteconfig-day-two.go

```diff
@@ -0,0 +1,101 @@
+package tests
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/eco-goinfra/pkg/reportxml"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/gitopsztp/internal/gitdetails"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/gitopsztp/internal/helper"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/gitopsztp/internal/tsparams"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/version"
+
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+)
+
+var _ = Describe("ZTP Siteconfig Operator's Day 2 configuration Test",
+	Label(tsparams.LabelSiteconfigDayTwoConfigTestCase), func() {
+
+		// These tests use the hub and spoke architecture.
+		BeforeEach(func() {
+			By("verifying that ZTP meets the minimum version")
+			versionInRange, err := version.IsVersionStringInRange(RANConfig.ZTPVersion, "4.17", "")
+			Expect(err).ToNot(HaveOccurred(), "Failed to compare ZTP version string")
+
+			if !versionInRange {
+				Skip("ZTP Siteconfig operator tests require ZTP 4.17 or later")
+			}
+
+		})
+
+		AfterEach(func() {
+			// Remove newly added custom label from the ClusterInstance CR underneath “extraLabels” field
+			// after the spoke cluster deployed using git flow.
+			By("resetting the clusters app back to the original settings")
+			err := gitdetails.SetGitDetailsInArgoCd(
+				tsparams.ArgoCdClustersAppName, tsparams.ArgoCdAppDetails[tsparams.ArgoCdClustersAppName],
+				true, false)
+			Expect(err).ToNot(HaveOccurred(), "Failed to reset clusters app git details")
+
+			// Make sure the newly added cluster label removed from ClusterInstance CR on hub cluster using the command.
+			// $ oc get clusterinstance <spoke cluster name> -n <spoke namespace>
+			// -o jsonpath='{.spec.extraLabels.ManagedCluster}' | jq
+			By("checking newly added cluster label removed from cluster instance CR")
```

Comment:
Since AfterEach gets executed after Skip is called, could you add an early return like in #253? Otherwise we may be running the checks on ClusterInstance labels even if this test doesn't apply, potentially causing failures

Reply:
Okay, updated. Thanks! the change will be in place after submitting a new patch for review.